modules/git/ref.go

```diff
@@ -84,6 +84,19 @@ func RefNameFromCommit(shortName string) RefName {
 	return RefName(shortName)
 }
 
+func RefNameFromTypeAndShortName(tp RefType, shortName string) RefName {
```

Comment:
NOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO

DO NOT DO THAT

Reply:
If you insist that it is right, you could open a new PR to refactor existing code (for example: `/tree-list`) to use the new mechanism first.

Reply:
This is reverted.